src/ema_filter.c

```diff
@@ -51,10 +51,12 @@ void ema_filter_update(EMAFilter *filter, float target, float dt) {
     filter->speed += dt * filter->accel;
 
     uint16_t speed_limit;
-    if (sign(filter->speed) == sign(filter->value)) {
-        speed_limit = filter->on_speed;
-    } else {
+    if ((filter->value * target >= 0) && (fabsf(filter->value) > fabsf(target))) {
```

Comment:
Could you please remove the extra parentheses around the comparison expressions? They are redundant and this is a very common case where the operator precedence should be clear enough. I don't mind explicit parentheses in uncommon/unclear situations but here they IMO just add visual clutter.